### ISSUE TITLE
Segfault in mutt_index_menu on syncing IMAP mailbox

### ISSUE THREAD

**Issue by l29ah** [neomutt/neomutt#1799]
```
(gdb) bt
#0  0x0000563652515ffb in mutt_index_menu () at index.c:1926
#1  0x0000563652525dfe in main (argc=1, argv=<optimized out>, envp=<optimized out>) at main.c:1233
```

If NeoMutt crashed, did your OS create a 'coredump' file?

yea

## NeoMutt Version

```
NeoMutt 20180716-2290-2adeab
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.1.18+ (x86_64)
ncurses: ncurses 6.1.20190609 (compiled with 6.1.20190609)
libidn: 1.35 (compiled with 1.35)
GPGme: 1.13.0
hcache backends: bdb

Compiler:
Using built-in specs.
COLLECT_GCC=x86_64-pc-linux-gnu-gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-pc-linux-gnu/8.3.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /var/tmp/portage/sys-devel/gcc-8.3.0-r1/work/gcc-8.3.0/configure --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --prefix=/usr --bindir=/usr/x86_64-pc-linux-gnu/gcc-bin/8.3.0 --includedir=/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include --datadir=/usr/share/gcc-data/x86_64-pc-linux-gnu/8.3.0 --mandir=/usr/share/gcc-data/x86_64-pc-linux-gnu/8.3.0/man --infodir=/usr/share/gcc-data/x86_64-pc-linux-gnu/8.3.0/info --with-gxx-include-dir=/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/include/g++-v8 --with-python-dir=/share/gcc-data/x86_64-pc-linux-gnu/8.3.0/python --enable-languages=c,c++,fortran --enable-obsolete --enable-secureplt --disable-werror --with-system-zlib --enable-nls --without-included-gettext --enable-checking=release --with-bugurl=https://bugs.gentoo.org/ --with-pkgversion='Gentoo Hardened 8.3.0-r1 p1.1' --enable-esp --enable-libstdcxx-time --disable-libstdcxx-pch --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-multilib --with-multilib-list=m32,m64 --disable-altivec --disable-fixed-point --enable-targets=all --enable-libgomp --disable-libmudflap --disable-libssp --disable-libmpx --disable-systemtap --enable-vtable-verify --enable-lto --without-isl --enable-default-pie --enable-default-ssp
Thread model: posix
gcc version 8.3.0 (Gentoo Hardened 8.3.0-r1 p1.1) 

Configure options: --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --docdir=/usr/share/doc/neomutt-99999999 --libdir=/usr/lib64 --disable-doc --enable-nls --disable-notmuch --enable-gpgme --enable-pgp --disable-smime --enable-bdb --disable-gdbm --disable-kyotocabinet --disable-qdbm --disable-tokyocabinet --enable-idn --disable-gss --disable-lmdb --enable-sasl --with-ui=ncurses --sysconfdir=/etc/neomutt --enable-ssl --disable-gnutls

Compilation CFLAGS: -O2 -pipe -march=ivybridge -mtune=skylake -ftree-vectorize -Og -ggdb3 -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR -I/include -I/usr/include/

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens +getaddrinfo 
  -gnutls +gpgme -gss +hcache -homespool +idn +inotify -locales_hack -lua +meta 
  -mixmaster +nls -notmuch +openssl +pgp +sasl -smime +start_color 
  +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc/neomutt"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

**Comment by N8Fear** [comment]
I guess I hit the same error with the current git head. It seems like Context->mailbox is the issue (as that is where the error happens. Shortly before it happens there is an error about "Mailbox closed" (coming from imap/command.c). 
If I comment out line 163 in command.c (mx_fastclose_mailbox(adata->mailbox);) the error message still pops up, but the crash is worked around. I guess I create other trouble, but that seems to be where to look for the error.

**Comment by flatcap** [comment]
I've added a lot more checks for `Context->mailbox`.
Unlike the others, the bug you found comes after a `Context->mailbox` check -- I overlooked the fact that some functions will kill the `Context`.

Fixed in 6873cc9dc
